src/txn/txn.c

```diff
@@ -605,10 +605,13 @@ __txn_commit_timestamp_validate(WT_SESSION_IMPL *session)
 		WT_RET_MSG(session, EINVAL, "no commit_timestamp required and "
 		    "timestamp set on this transaction");
```

Comment:
I am surprised there is no additional code checking the status of always/never and some transaction flags for the durable timestamp at this point in the code similar to what is there for commit and read.